ui-library-filesharing-ui-components/app/src/ChatComponents.tsx

```diff
@@ -0,0 +1,137 @@
+import {
+  usePropsFor,
+  MessageThread,
+  SendBox,
+  ActiveFileUpload,
+  FileMetadata,
+} from "@azure/communication-react";
+import React from "react";
+import axios from "axios";
+import Form from "form-data";
+import { v4 } from "uuid";
+
+function ChatComponents(): JSX.Element {
+  const allActiveFileUploads = React.useRef<ActiveFileUpload[] | []>([]);
+  const completedFileUploads = React.useRef<FileMetadata[] | []>([]);
+  const [files, setFiles] = React.useState<File[] | []>();
+  const [activeFileUploads, setActiveFileUploads] = React.useState<
+    ActiveFileUpload[] | []
+  >([]);
+
+  const updateFileUploadProgress = (
+    fileId: string,
+    progress: number,
+    complete: boolean = false
+  ) => {
+    const updatedData = allActiveFileUploads.current.map((active) => {
```

Comment:
This is a complex component, so every bit we can do to make it more readable is good.

Can you extract a free function `updateProgressForOneFile(fileUploads, fileId, progress, complete)` that just contains line 26 to 35?

Then this becomes

```
// A comment about why we need to update both a ref and state.
allActiveFileUploads.current = updateProgressForOneFile(allActiveFileUploads.current, fileId, proggress, complete);
setActiveFileUploads(allActiveFileUploads.current);
```
And the reader can focus on the state update rather than how the list is updated.

Reply:
Do you really think this makes it more readable :P
![image](https://user-images.githubusercontent.com/9782747/166747025-150506cf-54ff-4c39-a63c-421f52f0c95b.png)